minica/minica.go

```diff
@@ -0,0 +1,158 @@
+package minica
+
+import (
+	"crypto"
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	"go.step.sm/crypto/sshutil"
+	"go.step.sm/crypto/x509util"
+	"golang.org/x/crypto/ssh"
+)
+
+// MiniCA is the implementation of a simple X.509 and SSH CA.
+type MiniCA struct {
+	Root          *x509.Certificate
```

Comment:
Not quite sure if a CA should be immutable after initialization. Maybe not export these fields directly and instead export getters in case we do want for people to be accessing them?

Reply:
I think making these fields public adds some extra flexibility to the package without requiring some extra options. The user of this package can take precautions to avoid any problem.